check.sh

```diff
@@ -1,17 +1,19 @@
 #! /bin/bash
 
-shopt -s globstar nullglob
+# SC1091 - Don't complain about not being able to find files that don't exist.
+
+EXCLUSIONS=SC1091
 
 # print shellcheck version so we know what Github uses
 shellcheck -V
 
-# SC1091 - Don't complain about not being able to find files that don't exist.
-shellcheck --exclude=SC1091 \
+shellcheck --exclude=$EXCLUSIONS \
            ./setup/pi/setup-teslausb \
            ./pi-gen-sources/00-teslausb-tweaks/files/rc.local \
            ./run/archiveloop \
            ./run/auto.teslausb \
            ./run/remountfs_rw \
            ./run/send-push-message \
-           ./run/waitforidle \
-           ./**/*.{sh,ksh,bash}
+           ./run/waitforidle
+
+find . -type f -name '*.sh' -print0 | xargs -0 shellcheck --exclude=$EXCLUSIONS --external-sources "$@" || true
```

Comment:
Using find makes using extended globbing unnecessary. My own CI/CD was unable to parse it out (as does Travis).

Reply:
External sources is for shellcheck to follow them is you had them included in your scripts (did not validate if it is needed here, but JIC).

Reply:
Right, I know what the option does, but the various scripts will source other scripts that don't necessarily exist on the host that runs shellcheck, or that exist in different locations. For example, archiveloop sources /root/bin/setup-teslausb, which exists in the project, but not at that path, while run/rclone_archive/archive-clips.sh sources /root/.teslaCamRcloneConfig, which doesn't exist in the project at all.

Also, that `"$@"` in the `xargs` command line isn't needed I think (and probably unwanted, since it evaluates to whatever arguments the `check.sh` script itself was called with), and doesn't the `|| true` at the end make it so any shellcheck errors are ignored?

Reply:
You might be right. It is that there will be a "true"either way. Replace it with a "&&" or just drop the whole thing.